advanced/Templates/pihole-FTL.service

```diff
@@ -42,10 +43,11 @@ start() {
     chown pihole:pihole /var/log/pihole-FTL.log /var/log/pihole.log
     chmod 0644 /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
     echo "nameserver 127.0.0.1" | /sbin/resolvconf -a lo.piholeFTL
-    if setcap CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN+eip "$(which pihole-FTL)"; then
-      su -s /bin/sh -c "/usr/bin/pihole-FTL" "$FTLUSER"
-    else
-      echo "Warning: Starting pihole-FTL as root because setting capabilities is not supported on this system"
+    # capsh --keep=1 --secbits=0x04 enables SECBIT_NO_SETUID_FIXUP preventing the kernel from
+    # stripping capabilities when changing from UID=0 to UID=nonzero
+    capsh --keep=1 --secbits=0x04 --caps="CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN+eip" -- -c "runuser -u ${FTLUSER} ${FTLBINARY}"
```

Comment:
There appears to be a `--user` option for `capsh`, is that portable?

Reply:
It doesn't seem to work as expected.
```
root@vultr:~# capsh --keep=1 --secbits=0x04 --caps="CAP_NET_BIND_SERVICE,CAP_NET_RAW,CAP_NET_ADMIN+eip" --user="pihole" -c "/usr/bin/pihole-FTL"
Unable to set group list for user: Operation not permitted
```

Reply:
Okay, for this to work we have to add `CAP_SETUID,CAP_SETGID`